src/main/java/com/google/api/codegen/transformer/go/GoSurfaceNamer.java

```diff
@@ -100,9 +100,31 @@ public String getLongRunningOperationTypeName(ModelTypeTable typeTable, TypeRef
 
   @Override
   public List<String> getDocLines(Method method, GapicMethodConfig methodConfig) {
-    String text = DocumentationUtil.getDescription(method);
-    text = lowerFirstLetter(text);
-    return super.getDocLines(getApiMethodName(method, methodConfig.getVisibility()) + " " + text);
+    return super.getDocLines(
+        putDocMethodName(
+            method.getSimpleName(),
+            DocumentationUtil.getDescription(method),
+            methodConfig.getVisibility()));
+  }
+
+  /**
+   * Docs in Go usually starts with the name of the method. Prepend the method name if the name
+   * isn't already the first word.
+   */
+  @VisibleForTesting
+  String putDocMethodName(String methodName, String doc, VisibilityConfig visibility) {
+    doc = doc.trim();
+    String firstWord = doc.trim();
+    for (int i = 0; i < firstWord.length(); i++) {
+      if (Character.isWhitespace(firstWord.charAt(i))) {
+        firstWord = firstWord.substring(0, i);
+        break;
+      }
+    }
+    if (firstWord.equalsIgnoreCase(methodName)) {
+      return doc;
```

Comment:
Argh, I can't reconstruct what I was thinking. I remember I was curious about what seemed like an inconsistency in the caller, but I can't find it now. I think `equalsIgnoreCase` makes more sense here, since it helps in case the comment has weird capitalization. Sorry for the flip-flop.

Also in case there is weird capitalization of the method in the comment, maybe instead of `return doc` we should set `doc = doc.substring(firstWord.length()).trim()` here and have the `return` below apply to both cases. I think that's a better consistency guarantee. See the additional test cases I suggested. WDYT?